docs/scaling.md

```diff
@@ -12,12 +12,37 @@ As of v3.0, the controller supports having a hot-standby for [High Availability]
 
 ## Vertically Scaling
 
-You can scale the controller vertically:
+You can scale the controller vertically in these ways:
 
-- If you have many workflows, increase `--workflow-workers` and `--workflow-ttl-workers`.
-- Increase both `--qps` and `--burst`.
+### Container Resource Requests
 
-You will need to increase the controller's memory and CPU.
+If you observe the Controller using its total request CPU or memory, you should increase those.
+
+### Adding Goroutines to Increase Concurrency
+
+If you have sufficient CPU you can take advantage of it with more goroutines:
+
+- If you have many Workflows and you notice they're not being reconciled fast enough, increase `--workflow-workers`.
+- If you're using `TTLStrategy` in your Workflows and you notice they're not being deleted fast enough, increase `--workflow-ttl-workers`.
+- If you're using `PodGC` in your Workflows and you notice the Pods aren't being deleted fast enough, increase `--pod-cleanup-workers`.
+
+>= v3.5
+
+- If you're using a lot of `CronWorkflows` and they don't seem to be firing on time, increase `--cron-workflow-workers`.
+
+### K8S API Client Side Rate Limiting
+
+The K8S client library rate limits the messages that can go out. The default values are fairly low. If you frequently see a message similar to this in the Controller log (issued by the library):
+
+`Waited for 7.090296384s due to client-side throttling, not priority and fairness, request: GET:https://10.100.0.1:443/apis/argoproj.io/v1alpha1/namespaces/argo/workflowtemplates/s2t`
+
+or for >= v3.5: a warning like this (could be any CR, not just `WorkflowTemplate`):
+
+`Waited for 7.090296384s, request:GET:https://10.100.0.1:443/apis/argoproj.io/v1alpha1/namespaces/argo/workflowtemplates/s2t`
+
+then assuming your K8S API Server can handle it:
+
+- Increase both `--qps` and `--burst`. The `qps` value indicates the average number of queries per second allowed by the K8S Client. The `--burst` value is the number of queries/sec the Client receives before it starts enforcing `qps`, so typically `--burst` > `qps`.
```

Comment:
```suggestion
- Increase both `--qps` and `--burst`. The `qps` value indicates the average number of queries per second allowed by the K8S Client. The `burst` value is the number of queries/sec the Client receives before it starts enforcing `qps`, so typically `burst` > `qps`.
```

- Consistently use `--` or not when describing the values.

Reply:
Is there a good piece of external documentation you might recommend here? Could add a sentence at the end as "See [x]() for more details".

the k8s docs themselves [don't seem to](https://kubernetes.io/search/?q=qps) document this very explicitly. the [godoc for the k8s client](https://pkg.go.dev/k8s.io/client-go/rest#Config) doesn't either 😕

Reply:
unfortunately, I don't really have a good link either